src/dlsproto/client/internal/SharedResources.d

```diff
@@ -247,5 +280,130 @@ public final class SharedResources
 
             return fiber;
         }
+
+        /**********************************************************************
+
+            Gets a single-shot timer.
+
+            Params:
+                period_s = seconds part of timer interval
+                period_ms = milliseconds part of timer interval
+                timer_dg = delegate to call when timer fires.
+
+            Returs:
+                ITimer interface to a timer to use during the request's lifetime.
+
+        **********************************************************************/
+
+        ITimer getTimer ( uint period_s, uint period_ms, void delegate ( ) timer_dg )
+        {
+            auto timer = this.acquired_timers.acquire(new Timer);
+            timer.initialise(period_s, period_ms, timer_dg);
+            return timer;
+        }
+
+        /**********************************************************************
+
+            Inteface to a timer to be used during the request's lifetime.
+
+        **********************************************************************/
+
+        interface ITimer
+        {
+            /// Starts the timer
+            void start ( );
+
+            /// Stops the timer
+            void stop ( );
```

Comment:
I wonder if `cancel` makes more sense. "Stop" sounds more like a stopwatch timer that can be started and stopped, to me.

Reply:
Good point.